### ISSUE TITLE
[DateRangePicker | React | localization | i18n | typescript]: Date Picker _ UI state not announced.

### ISSUE THREAD

**Issue by durgaprasad1995** [SAP/ui5-webcomponents#9860]
### Bug Description

This is an accessibility bug when user select the date in the date picker it's not announcing the state of it(it's behaving some times working some times not)
**JAWS Version 2024.2312.53 ILM**
**Observed Behavior:**

On accessing and activating a date within the date picker, the state of the date that the accessed date has been selected is not announced to the end user.

 

**Expected Behavior:**

State of the should be read out to screen reader users.
Providing state of the UI on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.
Here information as ‘Checked/selected’ should be read out when sub menu items are accessed.

State (expand/collapse) of the UI should be read out to screen reader users.

![image](https://github.com/user-attachments/assets/c833de91-bf67-46d5-b590-064b5aafd933)

### Affected Component

DateRangePicker

### Expected Behaviour

State of the should be read out to screen reader users.
Providing state of the UI on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.
Here information as ‘Checked/selected’ should be read out when sub menu items are accessed.

State (expand/collapse) of the UI should be read out to screen reader users.

### Isolated Example

https://sap.github.io/ui5-webcomponents-react/v2/?path=/docs/inputs-daterangepicker--docs

### Steps to Reproduce

1.Open the URL (https://sap.github.io/ui5-webcomponents-react/v2/?path=/docs/inputs-daterangepicker--docs)
2.Click on DateRangePicker 
3.Open JAWS or NVDA tool for screen reader
4.Select one date check whether it is announcing the date with selected
5. if you do multiple times it will not work (it's behaving adohoc so repeat the process multiple times)
...

### Priority

Medium

### UI5 Web Components Version

1.20.0

### Browser

Chrome

### Operating System

Windows

### Organization

SAP

### Declaration

- [X] I’m not disclosing any internal or sensitive information.

**Comment by didip1000** [comment]
Hey @SAP/ui5-webcomponents-topic-b,

The selected state doesn't get announced when the range is being selected, but once the selection is complete and the popover is re-opened the selected dates are labeled as "selected'. This is different from openui5 where the dates get announced as "selected" as the range is being selected. We need to check which behaviour is correct, and fix it here if it isn't. 

Best,
Diana P.

**Comment by durgaprasad1995** [comment]
Hi @didip1000  and @unazko  any update on this

**Comment by unazko** [comment]
Hi @durgaprasad1995,

The issue will be fixed for webc version 2 via the following PR: https://github.com/SAP/ui5-webcomponents/pull/9934
After that we could trigger a fix transport into 1.24 version.

**Comment by durgaprasad1995** [comment]
> webc version 2 via the followi

Okay @unazko  Thanks a lot once the fix is available please let me know the version
